IBPSA/Fluid/Sources/BaseClasses/PartialSource_Xi_C.mo

```diff
@@ -0,0 +1,97 @@
+within IBPSA.Fluid.Sources.BaseClasses;
+partial model PartialSource_Xi_C
+  "Partial component source with parameter definitions for Xi and C"
+
+  extends IBPSA.Fluid.Sources.BaseClasses.PartialSource;
+
+  parameter Boolean use_X_in = false
+    "Get the composition (all fractions) from the input connector"
+    annotation(Evaluate=true, HideResult=true, Dialog(tab="Advanced"));
+  parameter Boolean use_Xi_in = false
+    "Get the composition (independent fractions) from the input connector"
+    annotation(Evaluate=true, HideResult=true, Dialog(group="Conditional inputs"));
+  parameter Boolean use_C_in = false
+    "Get the trace substances from the input connector"
+    annotation(Evaluate=true, HideResult=true, Dialog(group="Conditional inputs"));
+  parameter Medium.MassFraction X[Medium.nX](
+    final quantity=Medium.substanceNames) = Medium.X_default
+    "Fixed value of composition"
+    annotation (Dialog(enable = (not use_X_in) and Medium.nXi > 0, group="Fixed inputs"));
+  parameter Medium.ExtraProperty C[Medium.nC](
+    final quantity=Medium.extraPropertiesNames) = fill(0, Medium.nC)
+    "Fixed values of trace substances"
+    annotation (Dialog(enable = (not use_C_in) and Medium.nC > 0, group="Fixed inputs"));
+  Modelica.Blocks.Interfaces.RealInput X_in[Medium.nX](
+    each final unit = "kg/kg",
+    final quantity=Medium.substanceNames) if use_X_in
+    "Prescribed boundary composition"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.RealInput Xi_in[Medium.nXi](
+    each final unit = "kg/kg",
+    final quantity=Medium.substanceNames) if use_Xi_in
+    "Prescribed boundary composition"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.RealInput C_in[Medium.nC](
+    final quantity=Medium.extraPropertiesNames) if use_C_in
+    "Prescribed boundary trace substances"
+    annotation (Placement(transformation(extent={{-140,-100},{-100,-60}})));
+
+initial equation
+  assert(not use_X_in or not use_Xi_in,
+    "Cannot use both X and Xi inputs, choose either of the two.");
+
+  if not use_X_in and not use_Xi_in then
+    Modelica.Fluid.Utilities.checkBoundary(Medium.mediumName, Medium.substanceNames,
+      Medium.singleState, true, X_in_internal, "Boundary_pT");
+  end if;
+
+equation
+  if use_X_in then
```

Comment:
Why is there only a test on `use_X_in` but not also on `use_Xi_in` as in the initial equation section above?

Reply:
I thought `checkBoundary` only checked whether the values of `X` sum up to one, but this is not the case. I'll fix this!